src/res_core.ml

```diff
@@ -2711,6 +2711,11 @@ and parseJsxChildren p =
     (true, [parsePrimaryExpr ~operand:(parseAtomicExpr p) ~noCall:true p])
   | _ -> (false, loop p [])
 
+and removeModuleNameFromPunnedFieldValue exp =
```

Comment:
Can you move this helper somewhere at the beginning of the file?
(Context, the mutually recursive functions are mostly concerned with actual parsing logic, not really transformations) 
Would maybe also be nice to have a comment above with what the transformation actually looks like.

Reply:
Done, thanks for the feedback!